pkg/lang/javascript/aws_runtime/_fs.ts

```diff
@@ -0,0 +1,163 @@
+const flatted = require('flatted')
+const _ = require('lodash')
+const path = require('path')
+import { Readable } from 'stream'
+import {
+    S3Client,
+    PutObjectCommand,
+    GetObjectCommand,
+    ListObjectsCommand,
+    ListObjectsCommandInput,
+    DeleteObjectCommand,
+    HeadObjectCommand,
+} from '@aws-sdk/client-s3'
+
+const payloadBucketPhysicalName = process.env.KLOTHO_S3_PREFIX + '{{.PayloadsBucketName}}'
+const targetRegion = process.env['AWS_TARGET_REGION']
+
+const userBucketPath = '/files'
+
+const s3Client = new S3Client({ region: targetRegion })
+
+const streamToString = (stream: Readable): Promise<string> =>
+    new Promise((resolve, reject) => {
+        const chunks: Uint8Array[] = []
+        stream.on('data', (chunk) => chunks.push(chunk))
+        stream.on('error', reject)
+        stream.on('end', () => resolve(Buffer.concat(chunks).toString('utf8')))
+    })
+
+async function getCallParameters(paramKey, dispatcherMode) {
+    let isEmitter = dispatcherMode === 'emitter' ? true : false
+    try {
+        const bucketParams = {
+            Bucket: payloadBucketPhysicalName,
+            Key: paramKey,
+        }
+        const result = await s3Client.send(new GetObjectCommand(bucketParams))
+
+        let parameters = ''
+        if (result.Body) {
+            parameters = await streamToString(result.Body as Readable)
+        }
+
+        if (isEmitter && Array.isArray(parameters)) {
+            // Emitters only have 1 parameter - the runtime saves an array, so we
+            // normalize the parameter
+            parameters = _.get(parameters, '[0]')
+            if (Array.isArray(parameters)) {
+                let paramPairs = _.toPairs(parameters)
+                paramPairs = paramPairs.map((x) => {
+                    if (_.get(x, '[1].type') == 'Buffer') {
```

Comment:
similarly, if this is for index bounds, I _think_ typescript can do it as just `if (x[1]?.type == 'Buffer')`